hollow/src/main/java/com/netflix/hollow/core/read/engine/object/HollowObjectTypeReadStateShard.java

```diff
@@ -314,24 +315,18 @@ private long fieldOffset(HollowObjectTypeDataElements currentData, int ordinal,
      * @param out
      * @return
      */
-    private static final ThreadLocal<char[]> chararr = new ThreadLocal<char[]>();
+    private static final ThreadLocal<char[]> chararr = ThreadLocal.withInitial(() -> new char[100]);
 
     private String readString(ByteData data, long position, int length) {
-        long endPosition = position + length;
-
-        char chararr[] = getCharArray();
-
-        if(length > chararr.length)
+        char[] chararr = HollowObjectTypeReadStateShard.chararr.get();
+        if (length > chararr.length) {
             chararr = new char[length];
-
-        int count = 0;
-
-        while(position < endPosition) {
-            int c = VarInt.readVInt(data, position);
-            chararr[count++] = (char)c;
-            position += VarInt.sizeOfVInt(c);
+        } else {
+            Arrays.fill(chararr, 0, length, '\0');
```

Comment:
Is this necessary?  Why can't we just leave it dirty?

Reply:
It needs to start as 0's because the loop in `VarInt` requires an empty array (as it accumulates from a starting point of 0). This code specifically requires the initial value to be a zero'd array. 
```
             output[count] = (output[count] << 7) | (b & 0x7f);
```

Alternative options might be:
- do the zero'ing inside `VarInt::readVIntsInto`
- perform `Arrays.fill()` _after_ the `new String` construction, so that we only fill exactly the length we used
- add some more comments explaining ; I added some comments to the javadoc on `::readVIntsInto` but it seems potentially error-prone.

LMK thoughts.

Reply:
I see it now!  Thank you.